scripts/runLinter.lean

```diff
@@ -17,26 +18,64 @@ def readJsonFile (α) [FromJson α] (path : System.FilePath) : IO α := do
 def writeJsonFile [ToJson α] (path : System.FilePath) (a : α) : IO Unit :=
   IO.FS.writeFile path <| toJson a |>.pretty.push '\n'
 
+open Lake
+
+/-- Returns the root modules of `lean_exe` or `lean_lib` default targets in the Lake workspace. -/
+def resolveDefaultRootModules : IO (Array Name) := do
+  -- load the Lake workspace
+  let (elanInstall?, leanInstall?, lakeInstall?) ← findInstall?
+  let config ← MonadError.runEIO <| mkLoadConfig { elanInstall?, leanInstall?, lakeInstall? }
+  let workspace ← MonadError.runEIO <| MainM.runLogIO (loadWorkspace config)
```

Comment:
```suggestion
  let some workspace ← loadWorkspace config |>.toBaseIO
    | throw <| IO.userError "failed to load Lake workspace"
```

This should not use `MainM.runLogIO`. See leanprover-community/mathlib4#17953.

Reply:
Ah, okay, this is cleaner. Thanks!